_rules/descriptive-link-5effbb.md

```diff
@@ -0,0 +1,292 @@
+---
+id: 5effbb
+name: Link in context is descriptive
+rule_type: atomic
+description: |
+  This rule checks that the accessible name of a link together with its context describe its purpose.
+accessibility_requirements:
+  wcag20:2.4.4: # Link Purpose (In Context)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgements:
+  authors:
+    - Carlos Duarte
+    - Marie Trudelle
+    - Ramit Garg
+---
+
+## Applicability
+
+This rule applies to any HTML or SVG element which
+
+- has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
+- is [included in the accessibility tree][]; and
+- has a non-empty ("") [accessible name][].
+
+## Expectation
+
+The [accessible name][] of each target element together with its [programmatically determined link context][] describe the purpose of the link.
+
+## Assumptions
+
+- This rule assumes that the purpose of the link is not ambiguous to users in general when seen in context on the web page, which is the exception mentioned in success criterion [2.4.4 Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html). If the link is ambiguous to users in general, users of assistive technologies are not at a disadvantage when viewing the link out of context which makes it more of a general user experience concern than an accessibility issue.
+- This rule assumes that all elements with the [semantic role][] of ['link'][link] are used as links.
+- This rule assumes that all links have the [semantic role][] of ['link'][link].
```

Comment:
I don't think we need that last assumption. If there is a link which does not have the correct semantic role, the rule is going to ignore it and pass/be inapplicable but that's fine.

The previous assumption is needed because `<div role="link">Foo</div>` fails the rule but not the SC. This assumption is not needed because `<a href="contact.html" role="none">Bar</a>` fails the SC but not the rule; but this is not a problem.